sled-agent/src/services.rs

```diff
@@ -1904,10 +1928,10 @@ impl ServiceManager {
                     if is_gimlet {
                         // Maghemite for a scrimlet needs to be configured to
                         // talk to dendrite
-                        smfh.setprop("config/dendrite", "true")?;
                         smfh.setprop("config/dpd_host", "[::1]")?;
                         smfh.setprop("config/dpd_port", DENDRITE_PORT)?;
                     }
+                    smfh.setprop("config/dendrite", "true")?;
```

Comment:
I wonder: there isn't a `is_softnpu` that could apply here, right?

Reply:
My understanding of this `ServiceType::Maghemite { mode } = { ... }` case is that we are exclusively talking about the `mg-ddm` service in the switch zone, and that the maghemite service in the GZ is managed differently somewhere else. With that in mind, I think we always want to be in dendrite mode here. If that's not the case (now I'm paging back into memory that the Canada region does use the kernel routing tables in transit mode), we should have some development-specific configuration for that case, as `is_gimlet` no longer accurately represents that intent.